### PR TITLE
Add annotation support for generating doc config.

### PR DESCRIPTION
**Pull request by fanzhangio**
- parse doc annotations with +kubebuilder:doc:note and +kubebuilder:doc:warning
- support description_warning and description_note in config.yaml

#130

### CODE REVIEW

**Comment by pwittrock** [comment]
@fanzhangio would you add a test for this to the `test.sh` script?

**Comment by fanzhangio** [comment]
@pwittrock  Sure, will do.

**Comment by fanzhangio** [comment]
@pwittrock  Done. Added test into `test.sh`

File: cmd/kubebuilder/docs/gen.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ func (g CodeGenerator) Execute(dir string) error {
 			}
 			sort.Strings(vs)
 			r.Version = vs[0]
+			if version[vs[0]] != nil {
+				anotations := version[vs[0]].DocAnnotation
```
Comment (anfernee) [inline cmd/kubebuilder/docs/gen.go]: annotations

File: cmd/kubebuilder/docs/gen.go [hunk 2]
```diff
@@ -103,5 +108,8 @@ resource_categories: {{ range $category := .Categories }}
   resources: {{ range $resource := $category.Resources }}
   - name: "{{ $resource.Kind }}"
     version: "{{ $resource.Version }}"
-    group: "{{ $resource.Group }}"{{ end }}
+    group: "{{ $resource.Group }}"{{ if $resource.Warning }}
+    description_warning: "{{ $resource.Warning }}"{{ end -}}{{ if $resource.Note }}
+    description_note: "{{ $resource.Note }}"{{ end -}}
```
Comment (anfernee) [inline cmd/kubebuilder/docs/gen.go]: these lines look a little bit weird.. can you explain?

File: test.sh [hunk 3]
```diff
@@ -500,6 +500,11 @@ function test_docs {
   diff docs/reference/includes $kb_orig/test/docs/expected/includes
   diff docs/reference/manifest.json $kb_orig/test/docs/expected/manifest.json
   diff docs/reference/config.yaml $kb_orig/test/docs/expected/config.yaml
+
+  header_text "testing doc annotations"
+  sed -i -e "s|// +kubebuilder:categories=foo,bar|// +kubebuilder:categories=foo,bar\n// +kubebuilder:doc:note=test notes message annotations\n// +kubebuilder:doc:warning=test warnings message annotations|" pkg/apis/insect/v1beta1/bee_types.go
```
Comment (anfernee) [inline test.sh]: why not updating the file directly. running the script will leave the tree dirty. I think `sed`ding it is very hacky.

File: cmd/kubebuilder/docs/gen.go (hunk 1 above)
Comment (fanzhangio) [reply cmd/kubebuilder/docs/gen.go]: Thanks

File: cmd/kubebuilder/docs/gen.go (hunk 2 above)
Comment (fanzhangio) [reply cmd/kubebuilder/docs/gen.go]: The generated config.yaml is not supported to be modified directly by user, so annotations like `// +kubebuilder:doc:warning` and `// +kubebuilder:doc:note` need to be supported.
As seen in
https://github.com/kubernetes-incubator/reference-docs/blob/master/gen-apidocs/generators/config.yaml

File: test.sh (hunk 3 above)
Comment (fanzhangio) [reply test.sh]: By far, the scripts is only used for dev purpose. A series of tests are going to be added along with new features. Will consider a better way for e2e test.

Comment (pwittrock) [reply test.sh]: @anfernee agreed this isn't is a total hack, but something we have relied upon while we setup better better infrastructure for testing.

We are working on the integration / e2e test infrastructure this Sprint, and will migrate these tests.

File: cmd/kubebuilder/docs/gen.go (hunk 2 above)
Comment (anfernee) [reply cmd/kubebuilder/docs/gen.go]: sorry i didn't make it clear. I mean the template looks weird. what you want is probably like this
```golang
{{- if $resource.Warning }}
  description_warning: "{{ $resource.Warning }}"
{{- end }}
```

File: test.sh (hunk 3 above)
Comment (pwittrock) [inline test.sh]: Would you add a test that has the annotations on 2 different resources (e.g. add an annotation to Ant as well).

Comment (anfernee) [reply test.sh]: you can do it with `i` or `a` command in `sed` with something like this:
```
sed -i '/type [a-zA-Z]* struct/ i NEWLINE' FILE_TO_MODIFY
```

File: cmd/kubebuilder/docs/gen.go (hunk 2 above)
Comment (fanzhangio) [reply cmd/kubebuilder/docs/gen.go]: Thank you very much! I will make a change in a separate PR.

**Pull request merged**